packages/vincent-contracts/src/facets/VincentUserFacet.sol

```diff
@@ -82,44 +88,10 @@ contract VincentUserFacet is VincentBase {
         VincentAppStorage.AppVersion storage newAppVersion =
             as_.appIdToApp[appId].appVersions[getAppVersionIndex(appVersion)];
 
-        // Check if the App Manager has disabled the App
-        if (!newAppVersion.enabled) revert LibVincentUserFacet.AppVersionNotEnabled(appId, appVersion);
-
-        // Check if all registered tools for this app version are provided
-        if (
-            newAppVersion.toolIpfsCidHashes.length() > 0
-                && toolIpfsCids.length != newAppVersion.toolIpfsCidHashes.length()
-        ) {
+        if (newAppVersion.toolIpfsCidHashes.length() != toolCount) {
             revert LibVincentUserFacet.NotAllRegisteredToolsProvided(appId, appVersion);
         }
 
-        // Verify that every tool provided is registered and every registered tool is provided
-        if (toolIpfsCids.length > 0) {
-            bytes32[] memory providedToolHashes = new bytes32[](toolIpfsCids.length);
-            for (uint256 i = 0; i < toolIpfsCids.length; i++) {
-                providedToolHashes[i] = keccak256(abi.encodePacked(toolIpfsCids[i]));
-                if (!newAppVersion.toolIpfsCidHashes.contains(providedToolHashes[i])) {
-                    revert LibVincentUserFacet.ToolNotRegisteredForAppVersion(appId, appVersion, toolIpfsCids[i]);
-                }
-            }
```

Comment:
This was validating that every tool provided by the user to `permitAppVersion` is specified as a tool for the App version they're permitting. Without this, a User could provide any tools as long as the length matches the expected amount of tools for the App version